src/lang/cpp/internal/Parser.java

```diff
@@ -4278,9 +4280,11 @@ public int visit(ICPPASTClassVirtSpecifier classVirtSpecifier) {
 	@Override
 	public int visit(ICPPASTDecltypeSpecifier decltypeSpecifier) {
```

Comment:
Should this node have a `typ` parameter?

Reply:
it could help. I'll add that.

Reply:
No it seems that would be inconsistent:

```
public IConstructor Type_decltype(IConstructor $expression, ISourceLocation $loc, boolean $isMacroExpansion) {
      
    if (!$expression.getType().isSubtypeOf(_Expression)) {
      throw new IllegalArgumentException("Expected " + _Expression + " but got " + $expression.getType() + " for $expression:" + $expression);
    }
    
    Map<String, IValue> kwParams = new HashMap<String, IValue>();
    
    kwParams.put("src", $loc);
    
    
    if ($isMacroExpansion) kwParams.put("isMacroExpansion", vf.bool(true));
    return vf.constructor(_Type_decltype_1 , $expression).asWithKeywordParameters().setParameters(kwParams);
  }
  
  public IConstructor Type_float(ISourceLocation $loc, boolean $isMacroExpansion) {
    
    Map<String, IValue> kwParams = new HashMap<String, IValue>();
    
    kwParams.put("src", $loc);
    
    
    if ($isMacroExpansion) kwParams.put("isMacroExpansion", vf.bool(true));
    return vf.constructor(_Type_float_0 ).asWithKeywordParameters().setParameters(kwParams);
  }
```

So `float` does not have `typ=float()` to it yet either. If we would add that, then it would also make sense for the decltype. If you need it right now, you can get the `typ` off of the expression. 

So let's add this in a different PR if we need the TypeSymbols with every Type ast.